core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/hybrid/HybridRoleManager.java

```diff
@@ -1192,4 +1192,96 @@ private void throwRoleAlreadyExistsError(String roleName) throws UserStoreExcept
                 .getMessage(), roleName);
         throw new UserStoreException(errorCode + " - " + errorMessage, errorCode, null);
     }
+
+    /**
+     * Check whether the group exists in the UM_HYBRID_GROUP_ROLE table.
+     *
+     * @param groupName        The group name.
+     * @throws UserStoreException An unexpected exception has occurred.
+     */
+    public boolean isExistingGroup(String groupName) throws UserStoreException {
+
+        Connection dbConnection = null;
+        PreparedStatement prepStmt = null;
+        ResultSet rs = null;
+        boolean isExisting = false;
+
+        try {
+            dbConnection = DatabaseUtil.getDBConnection(dataSource);
+            prepStmt = dbConnection.prepareStatement(HybridJDBCConstants.GET_GROUP_ID);
+            prepStmt.setString(1, groupName);
+            prepStmt.setInt(2, tenantId);
+            rs = prepStmt.executeQuery();
+
+            if (rs.next()) {
+                int value = rs.getInt(1);
+                if (value > -1) {
+                    isExisting = true;
+                }
+            }
+            return !isExisting;
+        } catch (SQLException e) {
+            String errorMessage = "Error occurred while checking the existence of the group : " + groupName +
+                    "in hybrid group-role mapper";
+            if (log.isDebugEnabled()) {
+                log.debug(errorMessage, e);
+            }
+            throw new UserStoreException(errorMessage, e);
+        } finally {
+            DatabaseUtil.closeAllConnections(dbConnection, rs, prepStmt);
+        }
+    }
+
+    /**
+     * Update group name in the UM_HYBRID_GROUP_ROLE table.
+     *
+     * @param groupName        The current group name.
+     * @param newGroupName     The new group name.
+     * @throws UserStoreException An unexpected exception has occurred.
+     */
+    public void updateGroupName(String groupName, String newGroupName) throws UserStoreException {
+
+        if (this.isExistingGroup(groupName)) {
+            return;
+        }
+
+        Connection dbConnection = null;
+        try {
```

Comment:
Use a try-with-resource statement, it  ensures that each resource is closed at the end of the statement. Then no need to close the connection in finally block.
``` try (Connection dbConnection = DatabaseUtil.getDBConnection(dataSource)) {..}```
Fix other usage as well

Reply:
Fixed with ee2c01d7a4bbe760b4090f5652a8ae87cdbc987e